app/src/main/java/com/example/gpsapp/SecondActivity.java

```diff
@@ -13,7 +13,7 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_second);
 //Find your views
-       button2 = (Button) findViewById(R.id.button);
+       button2 = (Button) findViewById(R.id.button2);
```

Comment:
There is nothing in the second activity layout with the id of button. Changed it to button2 and that fixed the app crashing